### PR TITLE
Implementation of EventEmitter inside the dequeue

### PR DESCRIPTION
**Pull request by DarthPigrum**
It was stated as a task for this repository, so I've decided to try. My implementation has all methods that production EventEmitter from HPW/EventEmitter has but taken from EE of built-in library. .emit() is "protected" by holding in this._emitter which is the only way to say "don't touch" as I know. And actually I can't see better sollution because event holding it in wrapper has no sense(I think) cause this wrapper can't be sure from where the call comes. Please review my code and/or tell me if you know better way to protect the .emit() function or if I've implemented wrong contract for it.

### CODE REVIEW

File: JavaScript/3-dequeue.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Dequeue {
   constructor() {
     this.first = null;
     this.last = null;
+    this._emitter = new (require('events'));
```
Comment (tshemsedinov) [inline JavaScript/3-dequeue.js]: Don't use `require` inside constructor, this is obvious memory leak, you can move it to first lines of this file.

File: JavaScript/3-dequeue.js [hunk 2]
```diff
@@ -15,6 +16,7 @@ class Dequeue {
       this.first = element;
       this.last = element;
     }
+    this._emitter.emit('push',element.item,this)
```
Comment (tshemsedinov) [inline JavaScript/3-dequeue.js]: Use semicolons `;`

Comment (tshemsedinov) [reply JavaScript/3-dequeue.js]: and spaces after comma `,`

File: JavaScript/3-dequeue.js [hunk 3]
```diff
@@ -47,8 +51,30 @@ class Dequeue {
     } else {
       this.first = element.next;
     }
+    this._emitter.emit('shift',element.item,this)
     return element.item;
   }
+  on(...args){
```
Comment (tshemsedinov) [inline JavaScript/3-dequeue.js]:
```suggestion
  on(...args) {
```

**Review by tshemsedinov — COMMENTED** [review]
Use eslint before commit

**Comment by DarthPigrum** [comment]
Fixed

**Comment by tshemsedinov** [comment]
:+1:

**Pull request closed without merging**